src/test/runner.ts

```diff
@@ -167,7 +167,7 @@ export class Runner {
       testFiles.forEach(file => allTestFiles.add(file));
     }
 
-    const webServer = config.webServer && await WebServer.create(config.webServer);
+    const webServer = (!list && config.webServer) && await WebServer.create(config.webServer);
     let globalSetupResult: any;
     if (config.globalSetup)
```

Comment:
We are still running globalSetup when we are in this mode. Is there a reason globalSetup runs before the tests are listed? I would have expected it to run right before we partition tests into work.

@dgozman

Reply:
https://github.com/microsoft/playwright/issues/7029
https://github.com/microsoft/folio/pull/222